### DISCUSSION TITLE
bashrc vs bash_profile

### DISCUSSION THREAD

**Discussion by colinbes** [starship/starship#1161] [General]
On macOS (Catalina) if using bash and not zsh is correct file to add eval command to .bashrc or .bash_profile?

If I add to .bashrc (which is in my home folder) then I see no change to prompt when opening terminals, if I add to .bash_profile then I do see prompt change to starship inspired prompts.

**Comment by edubxb** [comment]
Usually, MacOs open Bash in "login mode", so it reads the `.bash_profile` file, in GNU/Linux (and other UNIXes I think), Bash is usually started in "interactive mode" (when using a terminal emulator), so it reads the `.bashrc` file.

As the Bash docs says:

> So, typically, your ~/.bash_profile contains the line
> 
> if [ -f ~/.bashrc ]; then . ~/.bashrc; fi

so you can achieve the expected behavior whatever the mode you use.

Bash docs: https://www.gnu.org/software/bash/manual/html_node/Bash-Startup-Files.html
